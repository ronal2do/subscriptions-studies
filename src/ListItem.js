// @flow
import React, { Component } from 'react'

type Props = {
  node: NodeType,
}

type NodeType = {
  name: string,
  id: string,
  avatarUrl: string,
}

export default class ListItem extends Component<Props> {
  render () {
    const { node: { name, id, avatarUrl } } = this.props
    return (
      <div
        className='card'
        key={id}
        style={{ backgroundImage: `url(${avatarUrl})` }}
      >
        <div className='card-container'>
          <p className='card-title'>{name}</p>
        </div>
      </div>
    )
  }
}
