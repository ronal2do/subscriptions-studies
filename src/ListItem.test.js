import React from 'react'
import ReactDOM from 'react-dom'
import ListItem from './ListItem'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<ListItem node={node} />', () => {
  const node = {
    name: 'my Title',
    id: '1',
    avatarUrl: 'https://pbs.twimg.com/profile_images/821713465245102080/mMtKIMax.jpg'
  }
  const wrapper = shallow(<ListItem node={node} />)

  it('should render ListItem', () => {
    // console.log(wrapper.debug())
  })

  it('should contain card classes', () => {
    expect(wrapper.find('.card').length).toBe(1)
    expect(wrapper.find('.card-container').length).toBe(1)
    expect(wrapper.find('.card-title').length).toBe(1)
  })

  it('should render props node', () => {
    expect(wrapper.instance().props.node).toBe(node)
  })

  it('should render props correct value value', () => {
    const tree = mount(<ListItem node={node} />)
    expect(tree.find('.card-title').text()).toBe('my Title')
    // expect(tree.find('style')).toBe({
    //   backgroundImage: `url(${node.avatarUrl})`
    // })

    tree.unmount()
  })

  it('Matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
