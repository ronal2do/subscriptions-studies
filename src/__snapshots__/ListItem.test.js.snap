// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem node={node} /> Matches the snapshot 1`] = `
<div
  className="card"
  key="1"
  style={
    Object {
      "backgroundImage": "url(https://pbs.twimg.com/profile_images/821713465245102080/mMtKIMax.jpg)",
    }
  }
>
  <div
    className="card-container"
  >
    <p
      className="card-title"
    >
      my Title
    </p>
  </div>
</div>
`;
